resources/qml/RoomDirectory.qml

```diff
@@ -0,0 +1,175 @@
+// SPDX-FileCopyrightText: 2021 Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+import QtQuick 2.9
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.3
+import im.nheko 1.0
+import im.nheko.RoomDirectoryModel 1.0
+
+ApplicationWindow {
+    id: roomDirectoryWindow
+    visible: true
+
+    x: MainWindow.x + (MainWindow.width / 2) - (width / 2)
+    y: MainWindow.y + (MainWindow.height / 2) - (height / 2)
+    minimumWidth: 650
+    minimumHeight: 420
+    palette: Nheko.colors
+    color: Nheko.colors.window
+    modality: Qt.WindowModal
+    flags: Qt.Dialog | Qt.WindowCloseButtonHint
+    title: qsTr("Explore Public Rooms")
+
+    Shortcut {
+        sequence: StandardKey.Cancel
+        onActivated: roomDirectoryWindow.close()
+    }
+
+    header: RowLayout {
+        id: searchBarLayout
+        spacing: Nheko.paddingMedium
+        width: parent.width      
+
+        implicitHeight: roomSearch.height
+
+        MatrixTextField {
+            id: roomSearch
+
+            Layout.fillWidth: true
+
+            font.pixelSize: fontMetrics.font.pixelSize
+            padding: Math.ceil(1.5 * Nheko.paddingSmall)

```

Comment:
It depends on the value of the padding. Currently it would be fine, because the padding is 4 and 1.5*4 = 6. But if we change it to 5, it would misalign and break font rendering on most systems. But really, this should probably be paddingMedium...